### PR TITLE
Implement Random Blocks transition

### PR DESCRIPTION
**Pull request by Albeleon**
Here it is. I left a note in both files (graphics.h and graphics.cpp explaining my choices).

### CODE REVIEW

**Comment by carstene1ns** [comment]
jenkins: ok to test

File: src/graphics.h [hunk 1]
```diff
@@ -18,6 +18,15 @@
 #ifndef EP_GRAPHICS_H
 #define EP_GRAPHICS_H
 
+// Constant Definitions
+#ifndef _SIZE_RANDOM_BLOCKS_
+#define _SIZE_RANDOM_BLOCKS_ 4
+#endif
+
+#ifndef _SIZE_ARRAY_RANDOM_BLOCKS_
+#define _SIZE_ARRAY_RANDOM_BLOCKS_ 4800
```
Comment (Ghabry) [inline src/graphics.h]: Please don't use #defines anymore, they are old-school. Also I wouldn't put them in the header.

The modern compile time constants are constexpr.
Just put them in a anonymous namespace (namespace { }) in graphics.cpp.
And follow the variable naming convention (random_blocks_size, random_blocks_array_size).

File: src/graphics.cpp [hunk 2]
```diff
@@ -60,6 +62,8 @@ namespace Graphics {
 
 	std::unique_ptr<MessageOverlay> message_overlay;
 	std::unique_ptr<FpsOverlay> fps_overlay;
+	
+	std::array<uint32_t, _SIZE_ARRAY_RANDOM_BLOCKS_> randomBlocks;
```
Comment (Ghabry) [inline src/graphics.cpp]: naming: random_blocks

File: src/graphics.cpp [hunk 3]
```diff
@@ -250,6 +258,10 @@ void Graphics::Transition(TransitionType type, int duration, bool erase) {
 			else
 				screen1 = screen2;
 		}
+		
+		if (type == TransitionRandomBlocks) {
+			std::shuffle(randomBlocks.begin(), randomBlocks.end(), std::default_random_engine(next_fps_time));
```
Comment (Ghabry) [inline src/graphics.cpp]: That's a clever solution!
For portability reasons (you can pass a seed via --seed command line argument) all random functions must use our shared RNG.
A suitable function here is probably Utils::GetRandomU32(), unfortunately the function is cpp-file static. You could add it to the utils.h header and then call it.

**Comment by Ghabry** [comment]
Oh and nice long git commit messages, you almost reach ChrisBreitweiser level of commit explanation xD

**Comment by Ghabry** [comment]
Test case in our TestGame: https://easyrpg.org/play/pr1368/?new-game&start-map-id=29&start-position=1%208 (talk with the person in front of you)

File: src/utils.h [hunk 4]
```diff
@@ -173,11 +174,11 @@ namespace Utils {
...
 
 	/**
-	* Gets a random number uniformly distributed in [0, U32_MAX] 
+	* Gets the Random Number Generator (RNG) from Utils.
 	*
-	* @return Random number in uint32_t range
+	* @return the random number generator
 	*/
-	uint32_t GetRandomU32();
+	std::mt19937 GetRNG();
```
Comment (Ghabry) [inline src/utils.h]: Should be returned by reference. Put  & after 19937.

**Comment by Ghabry** [comment]
Looks good to me now, next time you are in the chat I will tell you how to clean up the commit mess :)

**Comment by carstene1ns** [comment]
Related: #900.

**Pull request merged**